crates/networking/p2p/rlpx/connection.rs

```diff
@@ -159,49 +195,83 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
         match &self.state {
             RLPxConnectionState::Established(_) => {
                 info!("Started peer main loop");
+                // Wait for eth status message or timeout.
+                let _ = timeout(Duration::from_secs(15), self.wait_status_msg())
+                    .await
+                    .map_err(|_err| {
+                        RLPxError::HandshakeError("Timeout waiting for status msg".to_string())
+                    })?;
+                let mut broadcaster_receive = self.connection_broadcast_send.subscribe();
+                // Status message received, start listening for connections,
+                // and subscribe this connection to the broadcasting.
                 loop {
-                    match self.receive().await? {
-                        // TODO: implement handlers for each message type
-                        // https://github.com/lambdaclass/lambda_ethereum_rust/issues/1030
-                        Message::Disconnect(_) => info!("Received Disconnect"),
-                        Message::Ping(_) => info!("Received Ping"),
-                        Message::Pong(_) => info!("Received Pong"),
-                        Message::Status(_) => info!("Received Status"),
-                        Message::GetAccountRange(req) => {
-                            let response =
-                                process_account_range_request(req, self.storage.clone())?;
-                            self.send(Message::AccountRange(response)).await?
-                        }
-                        Message::GetBlockHeaders(msg_data) => {
-                            let response = BlockHeaders {
-                                id: msg_data.id,
-                                block_headers: msg_data.fetch_headers(&self.storage),
-                            };
-                            self.send(Message::BlockHeaders(response)).await?
-                        }
-                        Message::GetBlockBodies(msg_data) => {
-                            let response = BlockBodies {
-                                id: msg_data.id,
-                                block_bodies: msg_data.fetch_blocks(&self.storage),
-                            };
-                            self.send(Message::BlockBodies(response)).await?
+                    tokio::select! {
```

Comment:
In my [PR](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1136) I'm simplifying this function to extract the message handling in another function. 
That will be better for this update, I can merge this into mine once it get merged